Added rad2vec and deg2vec functionality to gdscript and visualscript

jxfaith opened the pull request:
Implemented and tested rad2vec and deg2vec functionality. I couldn't figure out how to get visual_script to draw lines in a test environment, so it remains untested there, but it worked in GDScript. The implementations are extremely similar, so I don't foresee the functions not working in visual_script.

I used the exact naming in the feature request. Considering that this is a 2d and 3d environment, it may better serve to specify "rad2vec2" and "deg2vec2" instead.

Implements #7263.

volzhs committed:
Show selected node in Scene dock when parent node is folded

Fix #7228

27thLiz committed:
Fix crash on project importing by dragging a folder.

Fixes #7226

27thLiz committed:
Particles2D: Fix initial size randomness property having no effect.

It was just a little typo :p

27thLiz committed:
TextureRegionEditor: Fix mouse wheel scroll speed.

Any given mouse wheel input will generate two InputEvents in godot.
The zoom methods here acted on both ones, effectively giving a step value of 4 instead of 2.
Fixes #7236

zaps166 committed:
WebM: Fix compilation on Linux/X11 ARM platform

neikeq committed:
KinematicBody: Fix wrong method bind return type

neikeq committed:
KinematicBody: Fix wrong method bind return type

jxfaith committed:
Added rad2vec and deg2vec functions
Fully implemented and tested in GDscript, implemented and visible when built for visual_script, but could not test.

jxfaith committed:
Added rad2vec and deg2vec functions
Fully implemented and tested in GDscript, implemented and visible when built for visual_script, but could not test.

jxfaith committed:
Added rad2vec and deg2vec functions
Fully implemented and tested in GDscript, implemented and visible when built for visual_script, but could not test.

jxfaith committed:
removed whitespace
removed an extra return added to the bottom of the source

jxfaith committed:
removed whitespace
removed an extra return added to the bottom of the source

jxfaith committed:
removed whitespace
removed an extra return added to the bottom of the source

jxfaith commented:
More references to the enum types than I initially noticed.

Any advice on how to catch issues like this prior to posting a pull request? My changes compiled fine locally, though I didn't check other build targets.

jxfaith committed:
Renamed functions, added missing enum case references
Now consistently named MATH_RAD2VEC and MATH_DEG2VEC.  Added missing case references to gd_functions.cpp.

jxfaith committed:
Renamed functions, added missing enum case references
Now consistently named MATH_RAD2VEC and MATH_DEG2VEC.  Added missing case references to gd_functions.cpp.

jxfaith committed:
Renamed functions, added missing enum case references
Now consistently named MATH_RAD2VEC and MATH_DEG2VEC.  Added missing case references to gd_functions.cpp.

jxfaith committed:
Fixed constructor
Fixed constructers in visual_script_builtin_funcs.cpp

jxfaith committed:
Fixed constructor
Fixed constructers in visual_script_builtin_funcs.cpp

jxfaith committed:
Fixed constructor
Fixed constructers in visual_script_builtin_funcs.cpp

bojidar-bg commented:
Well, I don't know if it is possible to get no errors from this, unless you are on a different compiler/version than the travis environment.

bojidar-bg commented:
Also, you would have to squash all those commits into one, in case you want @akien-mga to merge it :wink:

To do this, you might use `git rebase --interactive HEAD~<number-of-commits-to-change>` then follow instructions (and I guess, if something goes wrong or you just want to see if there is another way, there are plenty of git tutorials)

bojidar-bg commented on modules/gdscript/gd_functions.cpp:
```diff
@@ -381,6 +384,24 @@ void GDFunctions::call(Function p_func,const Variant **p_args,int p_arg_count,Va
+			v2x = Math::cos(*p_args[0]);
+			v2y = Math::sin(*p_args[0]);
+			Vector2 v(v2x, v2y);
+			r_ret = v;
```
Wouldn't `r_ret = Vector2(Math::cos(..), Math::sin(..))` work here as well?

jxfaith replied:
I used extra lines for clarity, but the compressed version does match the style better. Changed.

bojidar-bg commented on modules/gdscript/gd_functions.cpp:
```diff
@@ -107,6 +108,8 @@ const char *GDFunctions::get_func_name(Function p_func) {
 		"print_stack",
 		"instance_from_id",
+		"rad2vec",
+		"deg2vec",
```
That's incorrect -- those two should be right behind the name for "max" (otherwise, everything would go wild)

bojidar-bg commented on modules/visual_script/visual_script_builtin_funcs.cpp:
```diff
@@ -826,6 +835,24 @@ void VisualScriptBuiltinFunc::exec_func(BuiltinFunc p_func,const Variant** p_inp
+			v2x = Math::cos(*p_inputs[0]);
+			v2y = Math::sin(*p_inputs[0]);
+			Vector2 v (v2x,v2y);
+			*r_return = v;
```
Same as above, `r_return = Vector2(Math::cos(..), Math::sin(..))` might be way shorter and maybe a bit simpler.

bojidar-bg reviewed (changes requested):
Slight issues here and there though

jxfaith committed:
changes for review
fixed array declaration in gd_functions.cpp, compressed code in gd_functions.cpp and visualscript.cpp

jxfaith committed:
changes for review
fixed array declaration in gd_functions.cpp, compressed code in gd_functions.cpp and visualscript.cpp

jxfaith committed:
changes for review
fixed array declaration in gd_functions.cpp, compressed code in gd_functions.cpp and visualscript.cpp

jxfaith committed:
Merge branch 'master' of https://github.com/jxfaith/godot

jxfaith committed:
Merge branch 'master' of https://github.com/jxfaith/godot

jxfaith commented:
I appear to have duplicated all of my commits and broken the code in the process. Oh boy.

jxfaith commented:
I'm going to close and resubmit off of a cleaner fork. Something went wrong with my rebase.
